packages/flutter_tools/lib/src/base/io.dart

```diff
@@ -92,12 +93,34 @@ ExitFunction _exitFunction = _defaultExitFunction;
 
 /// Exits the process.
 ///
+/// Throws [StateError] if assertions are enabled and the dart:io exit
+/// is still active when called. This may indicate exit was called in
+/// a test without being configured correctly. This behavior can be
+/// removed by setting the `FLUTTER_TEST` environment
+/// variable to a non-null string.
+///
 /// This is analogous to the `exit` function in `dart:io`, except that this
 /// function may be set to a testing-friendly value by calling
 /// [setExitFunctionForTests] (and then restored to its default implementation
 /// with [restoreExitFunction]). The default implementation delegates to
 /// `dart:io`.
-ExitFunction get exit => _exitFunction;
+ExitFunction get exit {
+  assert(() {
+    if (_exitFunction == io.exit) {
+      try {
```

Comment:
Consider pulling the `expect()` check into a helper method, and simplifying here as in the last PR.

Reply:
Done